torch/lib/THD/master_worker/common/CommandChannel.cpp

```diff
@@ -53,6 +53,7 @@ MasterCommandChannel::MasterCommandChannel()
   std::tie(_port, world_size) = load_master_env();
 
   _sockets.assign(world_size, -1);
+  _mutexes = std::vector<std::mutex>(world_size);
```

Comment:
That is quite surprising to me. I thought that it will be copying happening right here, but it turns out compiler is smart enough to move it. Sorry!